src/test/java/org/cactoos/io/BytesOfTest.java

```diff
@@ -65,62 +69,62 @@ public void readsLargeInMemoryContent() throws Exception {
                     )
                 )
             ).asBytes().length,
-            Matchers.equalTo(body.length() * multiplier)
-        );
+            new IsEqual<>(body.length() * multiplier)
+        ).affirm();
     }
 
     @Test
     public void readsInputIntoBytes() throws Exception {
-        MatcherAssert.assertThat(
-            "Can't read bytes from Input",
-            new String(
+        new Assertion<>(
+            "must read bytes from Input",
+            new TextOf(
                 new BytesOf(
                     new InputOf("Hello, друг!")
-                ).asBytes(),
-                StandardCharsets.UTF_8
+                )
             ),
             Matchers.allOf(
-                Matchers.startsWith("Hello, "),
-                Matchers.endsWith("друг!")
+                new StartsWith("Hello, "),
+                new EndsWith("друг!")
             )
-        );
+        ).affirm();
     }
 
     @Test
     public void readsFromReader() throws Exception {
         final String source = "hello, друг!";
-        MatcherAssert.assertThat(
-            "Can't read string through a reader",
+        new Assertion<>(
+            "must read string through a reader",
             new TextOf(
-                new BytesOf(
-                    new StringReader(source),
-                    StandardCharsets.UTF_8,
-                    // @checkstyle MagicNumberCheck (1 line)
-                    16 << 10
+                new Sticky(
+                    new InputOf(
+                        new BytesOf(
```

Comment:
@victornoel why did you decorate `BytesOf` with `Sticky` and `InputOf`?

Reply:
@llorllale because `TextMatcherEnvelope` make things so that `asString()` is called multiple times on `TextOf` which reads multiple times on the `StringReader` which fails the second time because the stream is already closed.

Maybe we can fix this at the level of `TextMatcherEnvelope` instead? This could be a good idea actually.

Reply:
@llorllale I made an issue for this here: https://github.com/llorllale/cactoos-matchers/issues/132
If you assign it to me, I will solve it fast and we can rely on the new version of cactoos-matchers to avoid here the extra decorators.